examples/3_simple_storage.py

```diff
@@ -0,0 +1,176 @@
+#   (1) encrypted for a team or user (via the user's implicit self-team: e.g.
+# alice,alice),
+#   (2) persistent across logins
+#   (3) fast and durable.
+#
+# It supports putting, getting, listing, and deleting. There is also concurrency
+# support, but check out 5_secret_storage.py for that. A team has many
+# namespaces, a namespace has many entryKeys, and an entryKey has one current
+# entryValue. Namespaces and entryKeys are in cleartext, and the Keybase client
+# service will encrypt and sign the entryValue on the way in (as well as
+# decrypt and verify on the way out) so keybase servers cannot see it or forge
+# it.
+#
+# This example shows how you can use KVStoreClient to interact with the team
+# encrypted key-value store.
+###################################
+
+import asyncio
+import logging
+import os
+import sys
+from enum import Enum
+
+from pykeybasebot import Bot, EventType
+
+logging.basicConfig(level=logging.DEBUG)
+
+if "win32" in sys.platform:
+    # Windows specific event-loop policy
+    asyncio.set_event_loop_policy(
+        asyncio.WindowsProactorEventLoopPolicy()  # type: ignore
+    )
+
+
+class KVMsg(Enum):
+    PUT = "put"
+    GET = "get"
+    DELETE = "delete"
+    LIST = "list"
+    HELP = "help"
+
+
+class KVHandler:
+    """
+    KVHandler handles commands sent via chat to use the team key-value store.
+
+    KVHandler listens to chat messages of the form:
+    `!storage put <namespace> <key> <value> (<revision>)`
+    `!storage get <namespace> <key>`
+    `!storage delete <namespace> <key> (<revision>)`
+    `!storage list`  // list namespaces
+    `!storage list <namespace>`  // list entries in namespace
+    """
+
+    MSG_PREFIX = "!storage"
+
+    async def __call__(self, bot, event):
+        members_type = event.msg.channel.members_type
+        if not event.type == EventType.CHAT:
+            return
+
+        channel = event.msg.channel
+        user = event.msg.sender.username
+
+        # support teams and implicit self teams
+        team = channel.name
+        if members_type == "impteamnative" and channel.name == user:
+            team = "{0},{0}".format(channel.name)
+
+        msg = ""
+        try:
+            msg = event.msg.content.text.body.strip().split(" ")
+        except AttributeError:
+            return
+
+        if len(msg) < 2 or msg[0] != self.MSG_PREFIX:
+            return
+
+        action = msg[1]
+        if action == KVMsg.HELP.value:
+            return await self.handle_help(bot, channel, team, msg, action)
+        if action == KVMsg.LIST.value:
+            return await self.handle_list(bot, channel, team, msg, action)
+        if action == KVMsg.GET.value:
+            return await self.handle_get(bot, channel, team, msg, action)
+        if action == KVMsg.PUT.value:
+            return await self.handle_put(bot, channel, team, msg, action)
+        if action == KVMsg.DELETE.value:
+            return await self.handle_delete(bot, channel, team, msg, action)
```

Comment:
NAB (not a blocker): there's a pattern i've seen for this kind of construct in python where you make a dict with all your functions as the values, and then lookup based on the key. little less boiler-plate for adding a new one. nbd.

Reply:
did it~